proxy/proxy.go

```diff
@@ -1007,6 +966,33 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 	return response, nil
 }
 
+func (p *Proxy) getRoundTripper(ctx *context, req *http.Request) (http.RoundTripper, error) {
+	switch req.URL.Scheme {
+	case "fastcgi":
+		f := "index.php"
+		if sf, ok := ctx.StateBag()["fastCgiFilename"]; ok {
+			f = sf.(string)
+		}
+		rt, err := fastcgi.NewRoundTripper(p.log, req.URL.Host, f)
+		if err != nil {
+			return nil, err
+		}
+
+		// FastCgi expects the Host to be in form host:port
+		// It will then be split and added as 2 separate params to the backend process
+		if _, _, err := net.SplitHostPort(req.Host); err != nil {
```

Comment:
I would move back these last two statements with changing the request host and the remote address to the original scope, because it doesn't really belong to "getting a roundtripper"

Reply:
This is an adhoc for fastcgi, originally it was within fastcgi switch case as well

Reply:
i see :+1: